packages/ui-text-input/src/TextInput/index.tsx

```diff
@@ -140,6 +140,17 @@ class TextInput extends Component<TextInputProps, TextInputState> {
       })
     }
 
+    if (
+      getInteraction({ props: { interaction: prevProps.interaction } }) !==
+        'disabled' &&
+      getInteraction({ props: { interaction: this.props.interaction } }) ===
+        'disabled'
+    ) {
```

Comment:
IMO this should be either `props.interaction === 'disabled'`, if you want to check for the props only, or `getInteraction(props) === 'disabled'`, if you want to check globally. This util also checks for the html `disabled` prop (both `<TextInput disabled {...} />` and `<TextInput interaction="disabled" {...} />`), so I would pass the whole props object.

Reply:
Yeah, I might as well. I think the whole interaction util is overkill, but since the code has it everywhere when accessing value from `interaction` prop, I'd keep it for consistency's sake.